packages/keystatic/src/component-blocks/cloud-image-preview.tsx

```diff
@@ -10,65 +10,189 @@ import { Icon } from '@keystar/ui/icon';
+    alt?: string | number;
+  } = { src: '' }
+): ImageData {
+  return {
+    src: data.src,
+    alt: 'alt' in data && typeof data.alt === 'string' ? data.alt : '',
+    height: 'height' in data ? getDimension(data.height) : '',
+    width: 'width' in data ? getDimension(data.width) : '',
+  };
+}
+
+function Placeholder({ onChange }: { onChange: (data: ImageData) => void }) {
+  const [isFocused, setIsFocused] = useState(false);
+  const [pastedData, setPastedData] = useState<ImageData>(cleanImageData());
+  const [status, setStatus] = useState('');
 
   const config = useConfig();
 
   const onPaste = (event: React.ClipboardEvent<HTMLInputElement>) => {
+    event.preventDefault();
     const text = event.clipboardData.getData('text/plain');
     try {
-      const data: unknown = JSON.parse(text);
+      const data: any = JSON.parse(text);
       if (
         typeof data === 'object' &&
         data !== null &&
         'src' in data &&
         typeof data.src === 'string'
       ) {
-        event.preventDefault();
-        props.onChange({
-          src: data.src,
-          alt: 'alt' in data && typeof data.alt === 'string' ? data.alt : '',
-          height: 'height' in data ? getDimension(data.height) : '',
-          width: 'width' in data ? getDimension(data.width) : '',
-        });
+        setPastedData(cleanImageData(data));
         return;
       }
     } catch (err) {}
     const pattern = /^\s*!\[(.*)\]\(([a-z0-9_\-/:.]+)\)\s*$/;
     const match = text.match(pattern);
     if (match) {
-      event.preventDefault();
-      props.onChange({
-        src: match[2],
-        alt: match[1],
-        height: '',
-        width: '',
-      });
+      setPastedData(
+        cleanImageData({
+          src: match[2],
+          alt: match[1],
+        })
+      );
+      return;
+    }
+    setPastedData(cleanImageData({ src: text }));
+  };
+
+  useEffect(() => {
+    if (!pastedData || !pastedData.src) {
+      setStatus('');
+      return;
+    }
+    if (!isValidURL(pastedData.src)) {
+      setStatus('The pasted data is not a valid Image URL');
       return;
     }
+    const img = new Image();
+    setStatus('Checking pasted data...');
+    img.onload = () => {
+      onChange({
+        ...pastedData,
+        width: img.width.toString(),
+        height: img.height.toString(),
+      });
+    };
+    img.onerror = () => {
+      setStatus('The pasted data is not a valid Image URL');
+      console.log('Invalid Image URL pasted:', pastedData);
+    };
+    img.src = pastedData.src;
+    return () => {
+      img.onload = null;
+    };
+  }, [pastedData, onChange]);
+
+  return (
+    <NotEditable>
+      <VStack gap="regular">
+        <div style={{ position: 'relative' }}>
+          <input
+            type="text"
+            value=""
+            onFocus={() => setIsFocused(true)}
+            onBlur={() => setIsFocused(false)}
+            onClick={() => {
+              document.execCommand('paste');
```

Comment:
What is this doing? based on trying it on iOS, that's not how it works there? and [according to MDN, it's not at all supported](https://developer.mozilla.org/en-US/docs/Web/API/Document/execCommand#:~:text=or%20insertion%20point.-,paste,at%20the%20insertion%20point%20(replaces%20current%20selection).%20Disabled%20for%20web%20content.,-redo)  (except in browser extensions it seems)

Reply:
Weirdly, it seemed to work on my version of iOS, I may have been in a "trusted environment" the way I was previewing the build? although that doesn't make sense.

Alternatively (based on my on-device testing) the newer API (I think I was using `navigator.clipboard.readText()`) seemed to work on Safari Desktop but not iOS, whereas `execCommand` seemed to work on both.

Nevermind, I think we're going in a different direction anyway.